events/handler_test.go

```diff
@@ -0,0 +1,256 @@
+		ContainerName: "/root",
+	}
+	subDirectoryEvent := &Event{
+		ContainerName: "/root/subdir",
+	}
+
+	if checkIfIsSubcontainer(myRequest, sameDirectoryEvent) != true {
+		t.Errorf("should have found %v and %v had the same directory",
+			myRequest, sameDirectoryEvent)
+	}
+	if checkIfIsSubcontainer(myRequest, subDirectoryEvent) != false {
+		t.Errorf("should have found %v and %v had different directory",
+			myRequest, subDirectoryEvent)
+	}
+
+	myRequest.IncludeSubcontainers = true
+
+	if checkIfIsSubcontainer(myRequest, sameDirectoryEvent) != true {
+		t.Errorf("should have found %v and %v had the same directory",
+			myRequest.ContainerName, sameDirectoryEvent.ContainerName)
+	}
+	if checkIfIsSubcontainer(myRequest, subDirectoryEvent) != true {
+		t.Errorf("should have found %v was a subdirectory of %v",
+			subDirectoryEvent.ContainerName, myRequest.ContainerName)
+	}
+}
+
+func TestWatchEventsAddsEventsToEventManager(t *testing.T) {
+	myEventHolder, myRequest := initializeScenario(t)
+	myRequest.EventType[TypeOom] = true
+	outChannel := make(chan *Event, 10)
+
+	go func() {
+		myEventHolder.WatchEvents(outChannel, myRequest)
+	}()
+
+	fakeOom := createFakeOom(createOldTime(t))
+	fakeEvent := oomInstanceToEvent(fakeOom)
+	fakeOom2 := createFakeOom(time.Now())
+	fakeEvent2 := oomInstanceToEvent(fakeOom2)
+
+	myEventHolder.eventlist = append(myEventHolder.eventlist, fakeEvent)
+	for _, watcher := range myEventHolder.watchers {
+		watchRequest := watcher.request
+		inChannel := watcher.channel
+		if checkIfEventSatisfiesRequest(watchRequest, fakeEvent) {
+			inChannel <- fakeEvent
+		}
+	}
+
+	myEventHolder.eventlist = append(myEventHolder.eventlist, fakeEvent2)
+	for _, watcher := range myEventHolder.watchers {
+		watchRequest := watcher.request
+		inChannel := watcher.channel
+		if checkIfEventSatisfiesRequest(watchRequest, fakeEvent2) {
+			inChannel <- fakeEvent2
+		}
+	}
+
+	startTime := time.Now()
+	go func() {
+		i := 0
+		for i == 0 {
+			if time.Since(startTime) > (5 * time.Second) {
+				t.Errorf("Took too long to receive all the events")
+				close(outChannel)
+			}
+			time.Sleep(1 * time.Second)
+		}
+	}()
+
+	eventsFound := 0
+	go func() {
+		for event := range outChannel {
+			eventsFound += 1
+			if eventsFound == 1 {
+				if event != fakeEvent {
+					t.Errorf("Expected to find %v but found %v", fakeEvent, event)
+				}
+			} else if eventsFound == 2 {
+				if event != fakeEvent2 {
+					t.Errorf("Expected to find %v but found %v", fakeEvent, event)
+				}
+				close(outChannel)
+				break
+			} else {
+				t.Errorf("Wrong number of events found.")
+				close(outChannel)
+				break
+			}
+		}
+	}()
+}
+
+func TestAddEventAddsEventsToEventManager(t *testing.T) {
+	myEventHolder, _ := initializeScenario(t)
+	fakeOom := createFakeOom(createOldTime(t))
+	fakeEvent := oomInstanceToEvent(fakeOom)
+	myEventHolder.AddEvent(fakeEvent)
+	if myEventHolder.eventlist.Len() != 1 {
+		t.Errorf("Expected to return 1 events but received %v",
```

Comment:
use t.Fatal. That stops the test. Then below you can assume that we only have one element and thus not have the if statement